addon/mixins/route.js

```diff
@@ -177,12 +185,10 @@ const RouteMixin = Ember.Mixin.create({
 
     this.set('_infinityModelName', modelName);
 
-    this._ensureCompatibility();
-
     options = options ? assign({}, options) : {};
     const startingPage = options.startingPage === undefined ? 0 : options.startingPage-1;
-
     const perPage      = options.perPage || this.get('_perPage');
+    const store        = options.store && this.get(options.store) || this.get('store');
```

Comment:
We probably need to assert that `options.store` is a string in order to do `this.get`.  If they pass an object or something, it might blow up on them!